hail/src/main/c/Region.cpp

```diff
@@ -146,4 +170,23 @@ REGIONMETHOD(jlong, Region, nativeAllocate)(
   return reinterpret_cast<jlong>(r->region_->allocate((size_t)n));
 }
 
+REGIONMETHOD(void, Region, nativeAddReferenceTo)(
+  JNIEnv* env,
+  jobject thisJ,
+  jobject otherJ
+) {
+  auto r = static_cast<ScalaRegion*>(get_from_NativePtr(env, thisJ));
+  auto r2 = static_cast<ScalaRegion*>(get_from_NativePtr(env, otherJ));
+  r->region_->add_reference_to(r2->region_);
+}
+
+REGIONMETHOD(void, Region, nativeGetNewRegion)(
```

Comment:
so this destroys the region value memory associated with `thisJ` and replaces it with fresh memory?

Reply:
Not quite---it deletes the reference to the region associated with `thisJ` and replaces it with a fresh region.

If that region is referenced by other existing regions, the memory won't be freed until no other regions reference it.